test/test-packages.py

```diff
@@ -19,17 +19,20 @@ def main():
         packages = yaml.safe_load(f.read())
 
     def get_test_set():
+        installers = {'CONDA_NAME': 'container-conda-test.sh', 'PIP_NAME': 'container-script.sh'} 
         for package in packages['packages']:
-            package_main_name = re.findall(r'([\S]+)', package['PIP_NAME'])[0]
-            package_list = package['PIP_NAME']
-            package['main_name'] = package_main_name
-            py_script = package['PKG_TEST']
-            for container in ['amazon-linux2', 'centos8', 'centos8-py38', 'focal']:
-                yield (package_main_name, package_list, container, 'container-script.sh', py_script, container)
-            if 'APT_NAME' in package:
-                yield (package_main_name, package['APT_NAME'], 'focal', 'container-apt-test.sh', py_script, 'focal-apt')
-            if 'YUM_NAME' in package:
-                yield (package_main_name, package['YUM_NAME'], 'centos8', 'container-yum-test.sh', py_script, 'centos8-yum')
+            for install_type in installers.keys():
+                if install_type in package:
```

Comment:
I generally prefer doing something like this instead:
```
if install_type not in package:
    continue
```
I like this because it allows you to keep your main block of code from getting indented yet another level. Obviously this is just style, not functionality, so it's just a suggestion.